### ISSUE TITLE
Search deployments by name

### ISSUE THREAD

**Issue by jandegen** [amosproj/amos2023ss04-kubernetes-inventory-taker#61]
## User story
1. As a user
2. I want / need search deployments by their name
3. So that I can easily find them

## Acceptance criteria
- [ ] Results are displayed in a list view
- [ ] Search is triggered only after pressing enter or the corresponding icon
- [ ] The search is always queried against the database
- [ ] The search field shall be on the top right, right on top of the list
- [ ] The search field is displayed inside the "All Deployments" view

## Definition of done (DoD)
* Colours match with [color scheme](https://amos-p4.atlassian.net/l/cp/g5VzjL88)

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
